storage/libradb/src/lib.rs

```diff
@@ -538,8 +539,9 @@ impl LibraDB {
             })
             .collect::<Result<Vec<_>>>()?;
 
+        // TODO: Cache last epoch change version, to avoid a DB access in most cases.
         let client_epoch = self.ledger_store.get_epoch(client_known_version)?;
-        let validator_change_proof = if client_epoch < ledger_info_with_sigs.ledger_info().epoch() {
+        let validator_change_proof = if client_epoch < self.ledger_store.get_current_epoch()? {
```

Comment:
Why this change? Doesn't this lead to some race condition, since we now call `get_latest_ledger_info` twice in this function and they may return different version if there's a concurrent write?

Reply:
This changed because of the case that the latest LedgerInfo has Some(next_validator_set) hence the real current epoch number is latest_LI.epoch() + 1.

But it's a good point that the latest ledger_info can change when the request is running. We need to take that into consideration (not only this if-condition, but also skip returning info about epochs newer than the current epoch indicated by `ledger_version`.